src/ApacheOrcDotNet.OptimizedReader/Buffers/BaseColumnBuffer.cs

```diff
@@ -101,7 +81,90 @@ private protected void ReadBooleanStream(StreamDetail stream, ReadOnlySpan<byte>
                 return;
 
             var numSkipped = 0;
-            var bufferReader = new BufferReader(ResizeBuffer(stream, buffer, length));
+            var bufferReader = new BufferReader(GetDataStream(stream, buffer, length));
+            var numOfTotalBitsToSkip = stream.Positions.ValuesToSkip * 8 + stream.Positions.RemainingBits;
+            var numOfBytesToSkip = numOfTotalBitsToSkip / 8;
+            while (!bufferReader.Complete)
+            {
+                var numByteValuesRead = OptimizedByteRLE.ReadValues(ref bufferReader, _boolStreamBuffer);
+
+                for (int idx = 0; idx < numByteValuesRead; idx++)
+                {
+                    if (numSkipped++ < numOfBytesToSkip)
+                        continue;
+
+                    var decodedByte = _boolStreamBuffer[idx];
+                    var isFinalByte = bufferReader.Complete && idx >= numByteValuesRead - 1;
+
+                    // Skip remaining bits.
+                    if (numOfBytesToSkip % 8 != 0)
+                        decodedByte = (byte)(decodedByte << numOfTotalBitsToSkip % 8);
+
+                    if (isFinalByte && decodedByte == 0)
+                    {
+                        // Edge case where there is only one value for the row entry and that value is null
+                        outputValues[numValuesRead++] = false;
+                        return;
+                    }
+
+                    outputValues[numValuesRead++] = (decodedByte & 128) != 0;
+                    if (numValuesRead >= outputValues.Length)
```

Comment:
No, this needs to occur each time we add a new value to the output buffer (incrementing the `numValuesRead` variable), because it is possible to fill the output buffer before reaching the final byte.

Reply:
The usual condition is to add 8 bytes to the numValuesRead.  We only need to consider the edge case of less values being added when we're at the end of the input buffer.  For the non-edge-case we should be able to add 8 to numValuesRead and not check it at every bit.

Reply:
As per our last call, I kept this as it is.